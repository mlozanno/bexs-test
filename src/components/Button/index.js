import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ label }) => (
	<button type="button" data-testid="button">
		{label}
	</button>
);

Button.propTypes = {
	label: PropTypes.string.isRequired,
};

export default Button;
