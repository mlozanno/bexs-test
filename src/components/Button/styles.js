import styled from 'styled-components';
import { shade } from 'polished';

import { pxToRem, fontSize, colors } from '~/styles/theme';

export const StyledButton = styled.button`
	display: flex;
	grid-column: span 2;
	align-items: center;
	justify-content: center;
	height: ${pxToRem(50)};
	color: ${colors.light};
	font-weight: normal;
	font-size: ${fontSize.regular};
	text-align: center;
	text-transform: uppercase;
	background-color: ${colors.primary};
	border: none;
	border-radius: ${pxToRem(10)};
	transition: background-color 0.2s;

	> span {
		position: relative;
		transition: 0.5s;

		svg {
			position: absolute;
			top: calc(50% - ${pxToRem(10)});
			left: ${pxToRem(-30)};
		}

		&::after {
			position: absolute;
			top: 0;
			right: -20px;
			opacity: 0;
			transition: 0.5s;
			content: '\\00bb';
		}
	}

	&:hover {
		background-color: ${shade(0.25, colors.primary)};

		span {
			padding-right: 25px;

			&::after {
				right: 0;
				opacity: 1;
			}
		}
	}

	&:disabled {
		background: ${colors.grey};
		cursor: not-allowed;
		user-select: none;
		pointer-events: none;
	}

	.spin {
		animation: spin 2s infinite linear;
	}
	@keyframes spin {
		0% {
			transform: rotate(0deg);
		}
		100% {
			transform: rotate(359deg);
		}
	}
`;
