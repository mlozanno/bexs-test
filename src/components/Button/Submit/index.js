import React from 'react';
import PropTypes from 'prop-types';

import { StyledButton } from '../styles';

const SubmitButton = ({ icon, label, disabled, loading, ...rest }) => (
	<StyledButton
		data-testid="submit-button"
		disabled={disabled}
		{...rest}
		type="submit"
	>
		<span>
			{icon && loading && icon}
			{label}
		</span>
	</StyledButton>
);

SubmitButton.propTypes = {
	label: PropTypes.string.isRequired,
	icon: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
	disabled: PropTypes.bool,
	loading: PropTypes.bool,
};

SubmitButton.defaultProps = {
	icon: undefined,
	disabled: false,
	loading: false,
};

export default SubmitButton;
