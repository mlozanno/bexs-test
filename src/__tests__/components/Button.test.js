import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Button from '~/components/Button';

const buttonData = {
	label: 'test label',
	handleClick: jest.fn(),
};

describe('<Button />', () => {
	it('should be a function', () => {
		expect(typeof Button).toBe('function');
	});

	it('should has a label', () => {
		const { getByTestId } = render(<Button {...buttonData} />);

		const button = getByTestId('button');

		expect(button).toHaveTextContent('test label');
	});

	it('should has an onClick with handleClick function', () => {
		const { getByTestId } = render(<Button {...buttonData} />);

		const button = getByTestId('button');

		fireEvent.click(button);

		expect(buttonData.handleClick).toBeCalledTimes(1);
	});
});
