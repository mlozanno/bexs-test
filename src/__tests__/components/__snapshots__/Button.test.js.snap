// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be render component 1`] = `
<button
  data-testid="button"
  type="button"
>
  test label
</button>
`;
